File: graph/adjacency_list.go
package main

import "fmt"

type EdgeList struct {
	start int
	end int
	weight int
}

type Edges struct {
	node int
	weight int
}

func CreateAdjList(nodes int, edges []EdgeList) {
	graph := make(map[int][]Edges)

	for _, e := range edges {
		graph[e.start] = append(graph[e.start], Edges{e.end, e.weight})
		// graph[e.end] = append(graph[e.end], Edges{e.start, e.weight}) // uncomment if you want the graph to be undirected
	}

	for k,v := range graph {
		fmt.Print(k, " => ", v, "\n");
	}
}

func main(){
	nodes := 4
	edges := []EdgeList {
		{0, 1, 5},
		{0, 2, 3},
		{1, 2, 2},
		{2, 0, 1},
		{2, 3, 4},
		{3, 3, 7},
	}

	// direacted graph
	CreateAdjList(nodes, edges)
}

File: graph/adjacency_list.py
from collections import defaultdict

class Graph:
    def __init__(self, nodes):
        self.graph = defaultdict(list)

    def add_edge(self,i,j,weight):
        self.graph[i].append([j,weight]) # you can also using tuple instead of list
        # self.graph[j].append([i,weight]) # uncomment if you want the graph to be undirected

    def print(self):
        print(self.graph)

# directed graph
g = Graph(4)
g.add_edge(0,1,5)
g.add_edge(0,2,3)
g.add_edge(1,2,2)
g.add_edge(2,0,1)
g.add_edge(2,3,4)
g.add_edge(3,3,7)

g.print()
